> Could you please refer to CI breakage you're referring to and preferably include CI message about problem in commit message? Otherwise it makes not much sense. Also it would be really good to explain what was the reason for CI complaint really.

Here is the  CI msg from the PR #20082 :

FAILED: ccache /opt/sdk/zephyr-sdk-0.10.3/arc-zephyr-elf/bin/arc-zephyr-elf-gcc -DBUILD_VERSION=zephyr-v2.0.0-1505-g2fbb24ee3385 -DKERNEL -DPI_NUM_ITERATIONS=500 -D_FORTIFY_SOURCE=2 -D__ZEPHYR__=1 -I/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/kernel/include -I/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/arch/arc/include -I/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/include -I/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/include/drivers -Izephyr/include/generated -I/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/soc/arc/snps_nsim -I/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/subsys/testsuite/include -isystem /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/lib/libc/minimal/include -isystem /opt/sdk/zephyr-sdk-0.10.3/arc-zephyr-elf/bin/../lib/gcc/arc-zephyr-elf/8.3.0/include -isystem /opt/sdk/zephyr-sdk-0.10.3/arc-zephyr-elf/bin/../lib/gcc/arc-zephyr-elf/8.3.0/include-fixed -Os -imacros/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/sanity-out/nsim_hs/tests/kernel/fp_sharing/generic/kernel.fp_sharing.arc/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -mcpu=hs -imacros/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wno-unused-but-set-variable -Werror=implicit-int -Werror -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/tests/kernel/fp_sharing/generic=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/buildslave/src/github.com/zephyrproject-rtos/zephyr=ZEPHYR_BASE -ffunction-sections -fdata-sections -g3 -gdwarf-2 -fno-delete-null-pointer-checks -munaligned-access -mno-sdata -mmpy-option=plus_qmacw -mfpu=fpud_all -std=c99 -nostdinc -MD -MT CMakeFiles/app.dir/src/main.c.obj -MF CMakeFiles/app.dir/src/main.c.obj.d -o CMakeFiles/app.dir/src/main.c.obj   -c /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/tests/kernel/fp_sharing/generic/src/main.c
In file included from /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/tests/kernel/fp_sharing/generic/src/main.c:68:
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/tests/kernel/fp_sharing/generic/src/float_regs_arc_gcc.h: In function '_load_all_float_registers':
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/tests/kernel/fp_sharing/generic/src/float_regs_arc_gcc.h:41:8: error: unused variable 'temp' [-Werror=unused-variable]
  u32_t temp = 0;
        ^~~~
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/tests/kernel/fp_sharing/generic/src/float_regs_arc_gcc.h: In function '_store_all_float_registers':
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/tests/kernel/fp_sharing/generic/src/float_regs_arc_gcc.h:75:8: error: unused variable 'temp' [-Werror=unused-variable]
  u32_t temp = 0;
        ^~~~
cc1: all warnings being treated as errors
ninja: build stopped: subcommand failed.

--- a/tests/kernel/fp_sharing/generic/src/float_regs_arc_gcc.h
+++ b/tests/kernel/fp_sharing/generic/src/float_regs_arc_gcc.h
@@ -38,9 +38,9 @@
  */
 static inline void _load_all_float_registers(struct fp_register_set *regs)
 {
+#ifdef CONFIG_FP_FPU_DA
 	u32_t temp = 0;
 
-#ifdef CONFIG_FP_FPU_DA
 	__asm__ volatile (
 		"ld.ab %1, [%0, 4];\n\t"
 		"sr %1, [%2];\n\t"
@@ -72,9 +72,9 @@ static inline void _load_all_float_registers(struct fp_register_set *regs)
 
 static inline void _store_all_float_registers(struct fp_register_set *regs)
 {
+#ifdef CONFIG_FP_FPU_DA
 	u32_t temp = 0;
 
-#ifdef CONFIG_FP_FPU_DA
 	__asm__ volatile (
 		"lr %1, [%2];\n\t"
 		"st.ab %1, [%0, 4];\n\t"

